From 1e913f82e1d9838fd1aa7e4f32d92850cc21ae37 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Wed, 2 Nov 2022 17:56:00 +0100
Subject: [PATCH 023/529] mfd: stmfx: Remove .of_compatible from stmfx_cells
 for idd and ts

idd and ts features are not described in stmfx bindings. Remove the
.of_compatible field from relative mfd_cells to avoid having to add
corresponding disabled node in device trees using stmfx:
		stmfx_idd: idd {
			status = "disabled";
		};
		stmfx_ts: stmfx_ts {
			status = "disabled";
		};
Then, the warning "Failed to locate of_node [id: -1]" won't appear anymore.
.of_compatible could be added as soon as idd or ts bindings are described
and drivers available.

Change-Id: Id5518c2bef014d86d4d7cea2a05befa422ed43ba
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/273984
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Alexandre TORGUE <alexandre.torgue@foss.st.com>
---
 drivers/mfd/stmfx.c | 2 --
 1 file changed, 2 deletions(-)

--- a/drivers/mfd/stmfx.c
+++ b/drivers/mfd/stmfx.c
@@ -81,13 +81,11 @@ static struct mfd_cell stmfx_cells[] = {
 		.num_resources = ARRAY_SIZE(stmfx_pinctrl_resources),
 	},
 	{
-		.of_compatible = "st,stmfx-0300-idd",
 		.name = "stmfx-idd",
 		.resources = stmfx_idd_resources,
 		.num_resources = ARRAY_SIZE(stmfx_idd_resources),
 	},
 	{
-		.of_compatible = "st,stmfx-0300-ts",
 		.name = "stmfx-ts",
 		.resources = stmfx_ts_resources,
 		.num_resources = ARRAY_SIZE(stmfx_ts_resources),
